php/libraries/User.class.inc

```diff
@@ -492,8 +492,12 @@ class User extends UserPermissions
         \Password $password,
         ?\DateTime $expiry = null
     ): void {
+        if (password_verify($this->userInfo['Email'], (string) $password)) {
```

Comment:
This should've been added in https://github.com/aces/Loris/pull/5538 but was left out somehow.